File: utils/logger.py
import os
import datetime
import traceback

def log_error(error, command_name):
    now = datetime.datetime.now()
    date_dir = now.strftime("%Y-%m-%d")
    hour_file = now.strftime("%H-00.log")
    
    logs_dir = os.path.join("logs", date_dir)
    log_file_path = os.path.join(logs_dir, hour_file)
    
    if not os.path.exists(logs_dir):
        os.makedirs(logs_dir, exist_ok=True)
        
    timestamp = now.isoformat()
    stack = "".join(traceback.format_exception(type(error), error, error.__traceback__))
    
    log_entry = f"""
[{timestamp}] COMMAND: {command_name}
MESSAGE: {str(error)}
STACK: {stack}
--------------------------------------------------------------------------------
"""
    
    with open(log_file_path, "a", encoding="utf-8") as f:
        f.write(log_entry)
        
    return log_file_path

File: logger.js
const fs = require('fs');
const path = require('path');

/**
 * Logs an error to a file formatted by date and hour.
 * @param {Error} error The error object.
 * @param {string} commandName The name of the command that failed.
 * @returns {string} The path to the log file.
 */
function logError(error, commandName) {
    const now = new Date();
    const dateDir = `${now.getFullYear()}-${String(now.getMonth() + 1).padStart(2, '0')}-${String(now.getDate()).padStart(2, '0')}`;
    const hourFile = `${String(now.getHours()).padStart(2, '0')}-00.log`;

    const logsDir = path.join(__dirname, 'logs', dateDir);
    const logFilePath = path.join(logsDir, hourFile);

    // Create directories if they don't exist
    if (!fs.existsSync(logsDir)) {
        fs.mkdirSync(logsDir, { recursive: true });
    }

    const timestamp = now.toISOString();
    const logEntry = `
[${timestamp}] COMMAND: ${commandName}
MESSAGE: ${error.message}
STACK: ${error.stack}
--------------------------------------------------------------------------------
`;

    fs.appendFileSync(logFilePath, logEntry, 'utf8');

    // Return the relative path for the user
    return path.join('logs', dateDir, hourFile);
}

module.exports = {
    logError
};
